### PR TITLE
Update: Activities/{id}/zones  & Laps

### PR DESCRIPTION
**Pull request by lwasser**
this addresses the activities/{id}/zones endpoint (and will impact the lazy loaded option in /activities/{id}

addresses in part #506

### CODE REVIEW

File: src/stravalib/client.py [hunk 1]
```diff
@@ -1043,7 +1041,7 @@ def get_activity_zones(
         Returns
         -------
         :class:`list`
-            A list of :class:`stravalib.model.BaseActivityZone` objects.
```
Comment (lwasser) [inline src/stravalib/client.py:1046]: A rename to align with the spec

File: src/stravalib/model.py [hunk 2]
```diff
@@ -993,39 +991,44 @@ class ClubActivity(strava_model.ClubActivity):
...
 ):
     """
     Base class for activity zones.
 
-    A collection of :class:`stravalib.model.DistributionBucket` objects.
+    A collection of :class:`stravalib.strava_model.TimedZoneDistribution` objects.
     """
 
     # Field overrides from superclass for type extensions:
+    # TODO: what was the legacy behavior and is it ok to break it?
```
Comment (lwasser) [inline src/stravalib/model.py:1024]: @jsamoocha this is a sanity check. i will remove all comments once i'm clear this is correct. 

Is there legacy behavior that I need to consider here or can we just map to the API spec / returned values?

File: src/stravalib/model.py [hunk 3]
```diff
@@ -993,39 +991,44 @@ class ClubActivity(strava_model.ClubActivity):
...
     # Using type that is currently mimicking legacy behavior.
-    distribution_buckets: Optional[list[DistributionBucket]] = None  # type: ignore[assignment]
 
-    # TODO: ignoring type given legacy support will be deprecated
+    # This should return a list of TimedZoneDistribution objects
+    # But this returns an error
```
Comment (lwasser) [inline src/stravalib/model.py:1028]:
```suggestion
```

**Comment by lwasser** [comment]
ok ready for review! ✨ i'll clean up comments once the review is complete // before we merge !

File: src/stravalib/model.py [hunk 4]
```diff
@@ -652,10 +650,8 @@ class Lap(
     strava_model.Lap,
     BoundClientEntity,
 ):
-    # Field overrides from superclass for type extensions:
-    activity: Optional[MetaActivity] = None
-    athlete: Optional[Athlete] = None
 
+    # TODO: double check undocumented...
```
Comment (lwasser) [inline src/stravalib/model.py:654]:
```suggestion
```

File: src/stravalib/model.py [hunk 5]
```diff
@@ -993,39 +987,43 @@ class ClubActivity(strava_model.ClubActivity):
     pass
 
 
-class DistributionBucket(TimedZoneRange):
+# Should be TimedZoneDistribution
```
Comment (lwasser) [inline src/stravalib/model.py:990]:
```suggestion
```

File: src/stravalib/model.py [hunk 6]
```diff
@@ -652,9 +650,6 @@ class Lap(
     strava_model.Lap,
     BoundClientEntity,
 ):
-    # Field overrides from superclass for type extensions:
-    activity: Optional[MetaActivity] = None
-    athlete: Optional[Athlete] = None
```
Comment (jsamoocha) [inline src/stravalib/model.py:657]: These should remain here, as they're _our_ `MetaActivity` and `MetaAthlete` types (which inherit from `BoundClientEntity`)

File: src/stravalib/model.py (hunk 2 above)
Comment (jsamoocha) [reply src/stravalib/model.py:1019]: We override the `min` and `max` attributes to also allow floats as their type. You already took care of that.

**Review by jsamoocha — APPROVED** [review]
This looks good to merge, thanks!

**Pull request merged**